core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,242 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for solution.
+ */
+
+oppia.controller('StateSolution', [
+  '$scope', '$rootScope', '$modal', '$filter', '$injector',
+  'editorContextService', 'alertsService', 'INTERACTION_SPECS',
+  'stateSolutionService', 'explorationStatesService',
+  'StateSolutionHelperService', 'oppiaExplorationHtmlFormatterService',
+  'stateInteractionIdService', 'stateHintsService', 'UrlInterpolationService',
+  'SolutionObjectFactory', 'responsesService', 'AnswerClassificationService',
+  'explorationContextService', 'angularNameService',
+  function(
+    $scope, $rootScope, $modal, $filter, $injector,
+    editorContextService, alertsService, INTERACTION_SPECS,
+    stateSolutionService, explorationStatesService,
+    StateSolutionHelperService, oppiaExplorationHtmlFormatterService,
+    stateInteractionIdService, stateHintsService, UrlInterpolationService,
+    SolutionObjectFactory, responsesService, AnswerClassificationService,
+    explorationContextService, angularNameService) {
+    $scope.editorContextService = editorContextService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.StateSolutionHelperService = StateSolutionHelperService;
+    $scope.inlineSolutionEditorIsActive = false;
+    $scope.interactionHtml = '';
+    $scope.SOLUTION_EDITOR_FOCUS_LABEL = (
+      'currentInteractionHtmlForSolutionEditor');
+
+    $scope.stateHintsService = stateHintsService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.correctAnswer = null;
+
+    $scope.toggleInlineSolutionEditorIsActive = function() {
+      $scope.inlineSolutionEditorIsActive = (
+        !$scope.inlineSolutionEditorIsActive);
+    };
+
+    $scope.$on('stateEditorInitialized', function(evt, stateData) {
+      stateSolutionService.init(
+        editorContextService.getActiveStateName(),
+        stateData.interaction.solution);
+      $scope.interactionHtml = (
+        oppiaExplorationHtmlFormatterService.getInteractionHtml(
+          stateInteractionIdService.savedMemento,
+          explorationStatesService.getInteractionCustomizationArgsMemento(
+            editorContextService.getActiveStateName()),
+          $scope.SOLUTION_EDITOR_FOCUS_LABEL));
+
+      $scope.inlineSolutionEditorIsActive = false;
+      var interactionId = stateInteractionIdService.savedMemento;
+
+      $scope.currentInteractionCanHaveSolution = (
```

Comment:
Shouldn't this check happen "outside" this controller -- i.e. if it's false then the entire solution editor should not show at all and this controller isn't loaded?